experiments/storm/src/main/scala/io/gearpump/experiments/storm/processor/StormProcessor.scala

```diff
@@ -73,15 +78,21 @@ private[storm] class StormProcessor(taskContext : TaskContext, conf: UserConfig)
       })
     }
 
+    val topologyContext = buildTopologyContext(topology, stormConfig, taskToComponent.asJava, componentToSortedTasks,
+      componentToStreamFields, boltId, stormTaskId, bolt.isInstanceOf[ShellBolt])
+    outputCollector = new StormOutputCollector(taskContext, topologyContext)
     val delegate = new StormBoltOutputCollector(outputCollector)
-    val topologyContext = topologyContextBuilder.buildContext(pid, boltId)
     bolt.prepare(stormConfig, topologyContext, new OutputCollector(delegate))
   }
 
   override def onNext(msg: Message): Unit = {
     val stormTuple = msg.msg.asInstanceOf[StormTuple]
-    outputCollector.setTimestamp(msg.timestamp)
-    bolt.execute(stormTuple.toTuple(topologyContextBuilder))
+    val tuple = stormTuple.toTuple(topology, stormConfig, taskToComponent.asJava, componentToSortedTasks,
```

Comment:
stormTuple.toTuple looks like have the same dependency of TopologyContext?

```
buildTopologyContext(topology, stormConfig, taskToComponent.asJava, componentToSortedTasks,
+      componentToStreamFields, boltId, stormTaskId, bolt.isInstanceOf[ShellBolt])
```

Shoould we pass the Topologycontext directly?

Will every new tuple need a new TopologyContext?